numpy/__init__.pyi

```diff
@@ -358,6 +358,187 @@ class ndarray(Iterable, Sized, SupportsInt, SupportsFloat, SupportsComplex,
     def __getattr__(self, name) -> Any: ...
 
 
+_Scalar = TypeVar("_Scalar", bound="generic")
```

Comment:
I think we can actually define circular references in `.pyi` files, like I did for `_DtypeLike` above.

Reply:
This is the approach recommended by PEP484

Reply:
OK, in that case we should probably fix `_DtypeLike` :)

Reply:
To be clear, this is the recommended approach for typing `self`

Reply:
I was suggesting that `bound="generic"` could be written as `bound=generic`.

Reply:
I realize. I suppose the PEP self-typing example is not within a stub file, so perhaps my point is irrelevant.

Reply:
In typeshed we prefer not doing any quoting of this sort.